src/ksc/Ksc/SUF/AD.hs

```diff
@@ -521,37 +526,31 @@ sufFwdRevPassDef gst Def{ def_fun    = Fun JustFun f
                         , def_res_ty = t_ty
                         }
   = let
+      (rhs', bog_ty, mkRev) =
+        sufFwdRevPass gst (mkInScopeSet (patVars s)) rhs
+
       fwd = Def { def_fun    = Fun SUFFwdPass f
                 , def_pat    = s
                 , def_rhs    = UserRhs rhs'
                 , def_res_ty = TypeTuple [t_ty, bog_ty]
                 }
 
-      deltaVarsOfPat :: Pat -> S.Set Var
-      deltaVarsOfPat = S.fromList . map (tVarVar . deltaOfSimple) . patVars
-
-      (rhs', bog_ty, mkRev) =
-        sufFwdRevPass gst (mkInScopeSet (patVars s)) rhs
-
-      (_, lets_) = mkRev avoid (Var dt) (Var bog)
-
-      lets = foldr (\(p, er) rest -> Let p er . rest) id lets_
-
       rev = Def { def_fun    = Fun SUFRevPass f
                 , def_pat    = TupPat [ dt, bog ]
                 , def_rhs    = UserRhs rhs''
                 , def_res_ty = ds_ty
                 }
        where
-        rhs'' = lets (patToExpr (fmap deltaOfSimple s))
-
-      bog  = TVar bog_ty (Simple "bog_arg")
-      dt = TVar dt_ty (Simple "dt_arg")
-
-      ds_ty = tangentType (patType s)
-      dt_ty = tangentType t_ty
-
-      avoid = deltaVarsOfPat s `S.union` S.fromList (map tVarVar [dt, bog])
+        bog  = TVar bog_ty (Simple "bog_arg")
```

Comment:
Is there a note somewhere that says why this naming is ok?

Reply:
"bog_arg" - a reader might wonder if a unique name needed to be generated?

Reply:
Ah, I see. In this case it's because they are bound as function arguments. There's nothing in scope! On the other hand it might be clearer to use `notInScopeTVs` and explicitly pass in an empty scope.